tests/test_views.py

```diff
@@ -0,0 +1,36 @@
+from .test_base import BaseTestConfig
+from flask_login import current_user
+from models.models import User, db
+from flask import url_for
+import unittest
+
+class TestUserViews(BaseTestConfig):
+    
+    def test_index_page_loads(self):
+        response = self.client.get('/index')
+        # self.assertIn(response.data)
+
+    def test_correct_login(self):
+        #ensure that login works correctly with correct credentials 
+        with self.client:
+            response = self.client.post(
+                '/login',
+                data=dict(email="sample@example.com", password="just_test_user"),
+            )
+            self.assert_redirects(response, url_for('create_short'))
+        
+    def test_user_registration(self):
+        users = User.query.all()
+        with self.client:
+            response = self.client.post(
+                '/register',
+                data=dict(username="awesome", email="awesome@test.com", password="password",
+                password2="password"),
+                follow_redirects=True
+            )
+            new_count_users = User.query.all()
+            self.assertNotEqual(len(users), len(new_count_users))
```

Comment:
there is a better `assert` than asserting that the 2 are not equal. e.g your register url might actually be deleting a User so the new user count might be 1 User less than the old user count. This would mean your code is faulty but that would not be caught by this test because this test would pass in such a situation.

Assert that the new user count is an increment of 1 from the old user count..